tests/ci/setup.sh

```diff
@@ -25,54 +27,78 @@ install_dependencies() {
   PROCFILE_UTIL_PACKAGE_NAME="procfile-util_${PROCFILE_VERSION}_amd64.deb"
   curl -L "https://packagecloud.io/dokku/dokku/packages/ubuntu/trusty/procfile-util_${PROCFILE_VERSION}_amd64.deb/download.deb" -o "$ROOT_DIR/build/${PROCFILE_UTIL_PACKAGE_NAME}"
 
-  sudo add-apt-repository -y ppa:nginx/stable
-  sudo apt-get update
-  sudo apt-get -qq -y install nginx
-  sudo cp "${ROOT_DIR}/tests/dhparam.pem" /etc/nginx/dhparam.pem
+  sudo dpkg -i "${ROOT_DIR}/build/$HEROKUISH_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$PLUGN_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$SSHCOMMAND_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$SIGIL_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$PROCFILE_UTIL_PACKAGE_NAME"
+}
+
+build_dokku() {
+  echo "=====> build_dokku on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
+  "${ROOT_DIR}/contrib/release-dokku" build
+  docker build -t dokku/dokku:test .
+}
+
+run_dokku_container() {
+  echo "=====> run_dokku_container on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
+  docker run -d \
+    --env DOKKU_HOSTNAME=dokku.me \
+    --name dokku \
+    --publish 3022:22 \
+    --publish 22333:22 \
+    --publish 80:80 \
+    --publish 443:443 \
+    --volume /var/lib/dokku:/mnt/dokku \
+    --volume /var/run/docker.sock:/var/run/docker.sock \
+    dokku/dokku:test
+
+  check_container
+}
 
-  sudo dpkg -i "${ROOT_DIR}/build/$HEROKUISH_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$PLUGN_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$SSHCOMMAND_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$SIGIL_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$PROCFILE_UTIL_PACKAGE_NAME"
+check_container() {
```

Comment:
Shame there isn't a `SIGREADY` or some generic way to wait for a process to signal as such...